---
title: Что попадает в блок catch?
draft: false
tags:
  - "#JavaScript"
  - "#try-catch"
  - "#catch"
info:
---
Когда возникает ошибка, JavaScript генерирует объект, содержащий её детали.

Затем этот объект передаётся как аргумент в блок `catch`:

```javascript
try {   // ...
} catch(err) { // <-- объект ошибки, можно использовать другое название вместо err   // ... }`
```

Для всех встроенных ошибок этот объект имеет два основных свойства:
`name` Имя ошибки. Например, для неопределённой переменной это `"ReferenceError"`.
`message` Текстовое сообщение о деталях ошибки.

---

[[003 JSCore|Назад]]